### PR TITLE
initialize several unique_ptrs

### PR DESCRIPTION
**Pull request by neheb**
GCC's -fanalyzer complains that null pointers are created otherwise.

Some cleanup posibilities showed up.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/config/config_definition.cc [hunk 1]
```diff
@@ -130,7 +130,7 @@
 #define DEFAULT_MARK_PLAYED_ITEMS_STRING "*"
 
 /// \brief default values for CFG_IMPORT_SYSTEM_DIRECTORIES
-static std::vector<std::string_view> excludes_fullpath {
+static std::vector<std::string> excludes_fullpath {
```
Comment (KarlStraussberger) [inline src/config/config_definition.cc:133]: What's the problem with `string_view`?

Comment (neheb) [reply src/config/config_definition.cc:133]: Some problem in the constructor. Partial log: https://gist.github.com/neheb/73d3206c19a68ab3b2fd9d7bc35dd2d0

FWIW, this does not increase binary size. These strings are short enough to fit the short string optimization.

**Comment by neheb** [comment]
Added extra commit to mostly fix the rest of the unique_ptr warnings. Something tells me it shouldn't work.

File: src/content/update_manager.cc [hunk 2]
```diff
@@ -50,7 +50,7 @@ UpdateManager::UpdateManager(std::shared_ptr<Config> config, std::shared_ptr<Dat
     : config(std::move(config))
     , database(std::move(database))
     , server(std::move(server))
-    , objectIDHash(std::make_unique<std::unordered_set<int>>())
+    , objectIDHash(std::make_unique<std::unordered_set<int>>(MAX_OBJECT_IDS))
```
Comment (neheb) [inline src/content/update_manager.cc:53]: this is probably wrong.

**Comment by KarlStraussberger** [comment]
> Added extra commit to mostly fix the rest of the unique_ptr warnings. Something tells me it shouldn't work.

I'm waiting for this after creating 1.8.2.

**Comment by neheb** [comment]
OK. I split up the PR to keep the last one here.

**Pull request merged**